documentation/cluster-controller/cluster-controller.adoc

```diff
@@ -0,0 +1,296 @@
+worker nodes on startup. Default is `org.apache.kafka.connect.json.JsonConverter`
+* `KAFKA_CONNECT_KEY_CONVERTER_SCHEMAS_ENABLE`: if Kafka Connect transformation on keys are with or without schemas.
+It sets the `key.converter.schemas.enable` property in the properties configuration file used by Kafka Connect worker nodes on startup.
+Default is true
+* `KAFKA_CONNECT_VALUE_CONVERTER`: converter class used to convert values between Kafka Connect format and the serialized form 
+that is written to Kafka. It sets the `value.converter` property in the properties configuration file used by Kafka Connect 
+worker nodes on startup. Default is `org.apache.kafka.connect.json.JsonConverter`
+* `KAFKA_CONNECT_VALUE_CONVERTER_SCHEMAS_ENABLE`: if Kafka Connect transformation on values are with or without schemas.
+It sets the `value.converter.schemas.enable` property in the properties configuration file used by Kafka Connect worker nodes on startup.
+Default is true
+* `KAFKA_CONNECT_CONFIG_STORAGE_REPLICATION_FACTOR`: replication factor used when creating the configuration storage topic.
+It sets the `config.storage.replication.factor` property in the properties configuration file used by Kafka Connect worker nodes on startup.
+Default is 3
+* `KAFKA_CONNECT_OFFSET_STORAGE_REPLICATION_FACTOR`: replication factor used when creating the offset storage topic.
+It sets the `offset.storage.replication.factor` property in the properties configuration file used by Kafka Connect worker nodes on startup.
+Default is 3
+* `KAFKA_CONNECT_STATUS_STORAGE_REPLICATION_FACTOR`: replication factor used when creating the status storage topic.
+It sets the `status.storage.replication.factor` property in the properties configuration file used by Kafka Connect worker nodes on startup.
+Default is 3
+
+The following is an example of cluster configuration ConfigMap is the following.
+
+.Example Kafka Connect cluster ConfigMap
+[source,yaml,options="nowrap"]
+----
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: my-connect-cluster
+  labels:
+    strimzi.io/kind: cluster
+    strimzi.io/type: kafka-connect
+data:
+  nodes: "1"
+  image: "strimzi/kafka-connect:latest"
+  healthcheck-delay: "60"
+  healthcheck-timeout: "5"
+  KAFKA_CONNECT_BOOTSTRAP_SERVERS: "my-cluster-kafka:9092"
+  KAFKA_CONNECT_GROUP_ID: "my-connect-cluster"
+  KAFKA_CONNECT_KEY_CONVERTER: "org.apache.kafka.connect.json.JsonConverter"
+  KAFKA_CONNECT_KEY_CONVERTER_SCHEMAS_ENABLE: "true"
+  KAFKA_CONNECT_VALUE_CONVERTER: "org.apache.kafka.connect.json.JsonConverter"
+  KAFKA_CONNECT_VALUE_CONVERTER_SCHEMAS_ENABLE: "true"
+  KAFKA_CONNECT_CONFIG_STORAGE_REPLICATION_FACTOR: "3"
+  KAFKA_CONNECT_OFFSET_STORAGE_REPLICATION_FACTOR: "3"
+  KAFKA_CONNECT_STATUS_STORAGE_REPLICATION_FACTOR: "3"
+----
+
+In order to add more connectors plugins to the Kafka Connect deployment, it's possible to use the Stream2Image feature if
```

Comment:
This seems a bit out of context ... many people will have no clue what is Source2Image and how does it work.

Reply:
so ? should I delete it , or ? how do we want to introduce the S2I feature ?

Reply:
Well, if you ask me you should explain what it is and how it should be used. Or at least link to README which explains what it is.

TBH, I think this whole think should have started from the README.md. Because now you have two separate pieces of documentation. The first one (README.md) does the basics and does tzhem fairly well. The second does a bit more but doesn't really link with the first one. If you don't know about Strimzi and you get this ClusterController docu, you wouldn't know how to deploy Strimzi out of it.

We need to either link them together somehow or copy paste the things between them.

Reply:
I think that the README is something that should help you to start with the project while this cluster controller documentation should help you to understand a little bit more on how it works and how to specify the cluster configuration in the ConfigMaps.
I don't think that we should explain what Source2Image is in the cluster controller configuration.
Maybe we should have a different piece of documentation explaining how Kafka Connect S2I works even if you explained it better in the README.

Reply:
It doesn't make sense to maintain something in README and something in documentation. You want to have a single point + you want to be able to distribute the documentation. So IMHO we need to put everything into docu and have the docu on our website. And README can just link to the website.

Reply:
BTW: We can just ignore it and do it in next PR ... you don't have to do it here and now. ;-)

Reply:
But here we are talking to re-define the documentation structure which should not be done in this PR.

Reply:
Well ... I guess you can see it that way.

Reply:
I think you need a whole other section for it. Just slipping in a sentence here makes it too hard to discover that this functionality exists.